datafusion/sqllogictest/src/engines/datafusion_engine/runner.rs

```diff
@@ -69,9 +71,11 @@ impl sqllogictest::AsyncDB for DataFusion {
 
 async fn run_query(ctx: &SessionContext, sql: impl Into<String>) -> Result<DFOutput> {
     let df = ctx.sql(sql.into().as_str()).await?;
+    let task_ctx = Arc::new(df.task_ctx());
+    let plan = df.create_physical_plan().await?;
 
-    let types = normalize::convert_schema_to_types(df.schema().fields());
-    let results: Vec<RecordBatch> = df.collect().await?;
+    let types = normalize::convert_schema_to_types(plan.schema().fields());
+    let results: Vec<RecordBatch> = collect(plan, task_ctx).await?;
```

Comment:
Another thing you could do that might be even closer to the results, would be to call `execute` to get a `SendableRecordBatchStream` and retrieve the schema from that  (e.g `stream.schema()`) which should be exactly the same schema as RecordBatch produced

I think using the physical plan's schema is fine too though and better than what is on main

Reply:
It would be better to keep them the same.  Changed to using `SendableRecordBatchStream`. Thanks @alamb